packages/entity/src/internal/EntityDataManager.ts

```diff
@@ -192,7 +236,20 @@ export default class EntityDataManager<TFields> {
     fieldValues: readonly NonNullable<TFields[N]>[]
   ): Promise<void> {
     await this.entityCache.invalidateManyAsync(fieldName, fieldValues);
-    const dataLoader = this.getFieldDataLoaderForFieldName(fieldName);
+    const dataLoader = this.getRegularFieldDataLoderForFieldName(fieldName);
+    fieldValues.forEach((fieldValue) => dataLoader.clear(fieldValue));
+  }
+
+  private async invalidateManyByFieldEqualingTransactionalAsync<N extends keyof TFields>(
+    queryContext: EntityQueryContext,
+    fieldName: N,
+    fieldValues: readonly NonNullable<TFields[N]>[]
+  ): Promise<void> {
+    if (!queryContext.isInTransaction()) {
+      return;
+    }
```

Comment:
Instead of returning here, it seems better not to call into the transactional methods in the first place for slightly simpler call trees. Either way there's going to be an `inTransaction()` check somewhere, so doing it earlier and higher up in the call tree seems simpler.

```ts
    // EntityMutator.ts

    if (queryContext.inTransaction()) { // <-- new
      await Promise.all([
        entityLoader.invalidateFieldsTransactionalAsync(this.originalFieldsForEntity),
        entityLoader.invalidateFieldsTransactionalAsync(this.fieldsForEntity),
      ]);
    }
```